tools/github-actions/cascading/src/cascading.ts

```diff
@@ -0,0 +1,284 @@
+import * as Context from '@actions/github/lib/context';
+import * as github from '@actions/github';
+import {GitHub} from '@actions/github/lib/utils';
+
+const promisifiedExec = promisify(exec);
+
+/** Describes the minimal interface that a logger need to implement to be used by the Cascading plugin */
+export type BaseLogger = { debug: (log: string) => void; info: (log: string) => void; warning: (log: string) => void; error: (log: string) => void };
+
+/**
+ * Options expected by the Cascading plug-in
+ */
+export interface CascadingOptions<T extends BaseLogger> {
+  /** Logger, you can provide here core from @actions/core for example */
+  logger: T;
+
+  /** Force a merge commit in the cascading */
+  noFf: string;
+
+  /** The cascading pattern that needs to be applied */
+  cascadingPattern: string;
+
+  /** Whether to assign the first committer to the cascading failure PR */
+  assignCommitter: string;
+
+  /** The list of packages changes to ignore if the conflict is only about them (ex: ['@otter/common','@otter/core','@refx/booking-common','@refx/booking-components']) */
+  conflictsIgnoredPackages: string[];
+
+  /** Access token for GitHub API authentication */
+  token: string;
+
+  /** The default branch if you have one (ex: master, development), if no candidate found with the given pattern this branch will be the last one where the code will be cascaded */
+  defaultBranch: string;
+
+  /** Branch where the build is performed */
+  baseBranch: string;
+
+  /** Name of the build workflow to trigger build on cascaded branch */
+  buildWorkflow: string;
+
+}
+
+/**
+ * Handles the cascading to the next branch
+ */
+export class Cascading {
+  /** Name of the repository */
+  public repositoryName: string;
+
+  /** Repository owner login */
+  public ownerName: string;
+
+  constructor(public githubClient: InstanceType<typeof GitHub>, public githubContext: Context.Context, public options: CascadingOptions<BaseLogger>) {
+    this.ownerName = this.githubContext.payload.repository!.owner.login;
+    this.repositoryName = github.context.payload.repository!.name;
+    this.options.logger.info(`ownerName : ${this.ownerName}`);
+    this.options.logger.info(`repositoryName : ${this.repositoryName}`);
+  }
+
+  /**
+   * Launch the cascading process
+   */
+  public async execute() {
+    this.options.logger.info('Cascading plugin execution');
+    await handlePromisifiedExecLog(promisifiedExec('git config --global user.email "auto-cascading@amadeus.com"'), this.options.logger);
+    await handlePromisifiedExecLog(promisifiedExec('git config --global user.name "Auto cascading"'), this.options.logger);
+    await handlePromisifiedExecLog(promisifiedExec('git status'), this.options.logger);
+    const branchOutput = await handlePromisifiedExecLog(promisifiedExec(`git branch -al */${this.options.cascadingPattern}`), this.options.logger);
+    let sortedBranches = formatGitBranchOutput(branchOutput.stdout);
```

Comment:
it's the `verifyBranches` that seem to do the sorting.
Function and variable names aren't well chosen